traitsui/qt4/tree_editor.py

```diff
@@ -1058,16 +1058,20 @@ def _new_actions(self, node, object):
 
         for klass in add:
             prompt = False
+            factory = None
             if isinstance(klass, tuple):
-                klass, prompt = klass
+                klass, prompt, *factory_list = klass
+                if factory_list:
+                    factory = factory_list[0]
```

Comment:
I'd just do an explicit `if len(...) == 2: ... elif len(...) == 3: ...` here.  Clearer for people reading the code, I think.  You could also pull the `factory = None` into the length 2 case.

Reply:
Actually event better, you could do `factory = klass` in the length 2 case here.

Reply:
Oops, not also need to handle the case where we're just given a bare class.